README.md

```diff
@@ -44,65 +43,95 @@ Learn how to use Cloud Code for:
 #### Local development
 
 1. Set Project Id:
+
     ```bash
     export GOOGLE_CLOUD_PROJECT=<GCP_PROJECT_ID>
     ```
-2. Start the server with hot reload:
+
+2. Build and Start the server:
+
     ```bash
-    
+    go build -o server && ./server
     ```
 
 #### Deploying a Cloud Run service
 
 1. Set Project Id:
+
     ```bash
     export GOOGLE_CLOUD_PROJECT=<GCP_PROJECT_ID>
     ```
-2. Build the container using a buildpack:
+
+2. Use the gcloud credential helper to authorize Docker to push to your
+   Container Registry:
+
+   ```bash
+    gcloud auth configure-docker
+    ```
+
+3. Build and push the container using docker:
+
     ```bash
-    
+    docker build . -t gcr.io/$GOOGLE_CLOUD_PROJECT/microservice-template
```

Comment:
I am unable to run this docker image locally

Reply:
Hmm, this worked for me via docker. The command I used was `docker run --rm -it -p 8080:8080 -e GOOGLE_CLOUD_PROJECT=$GOOGLE_CLOUD_PROJECT gcr.io/$GOOGLE_CLOUD_PROJECT/microservice-template`. Does that command work, or what error are you getting?

Reply:
Ah I wasn't passing in the `$GOOGLE_CLOUD_PROJECT` as an env var to docker run. It doesn't seem to error like it's suppose to. After about maybe 30s I see the error: `2021/06/15 23:55:54 unable to initialize application: unable to detect Project ID from GOOGLE_CLOUD_PROJECT or metadata server: compute: Received 500 `host unreachable`

Reply:
One of the patterns we often use is to skip metadata server requests for the current project if that variable is set.